readme.md

```diff
@@ -46,13 +46,85 @@ unsubscribe();
 
 ### exitHook(onExit)
 
-Returns a function that removes the hook when called.
+Register a function to run during `process.exit`. Returns a function that removes the hook when called.
 
 #### onExit
 
 Type: `Function`
 
-The callback function to execute when the process exits.
+Describes a callback to run on `process.exit`.
+
+### asyncExitHook(onExit, minimumWait)
+
+Register a function to run during `gracefulExit`. Returns a function that removes the hook when called.
+
+Please see [Async Notes](#async-notes) for considerations when using the asynchronous API.
+
+```js
+import {asyncExitHook} from 'exit-hook';
+
+asyncExitHook(async () => {
+	console.log('Exiting');
+}, 300);
+
+throw new Error('🦄');
+
+//=> 'Exiting'
+```
+
+Removing an asynchronous exit hook:
+
+```js
+import {asyncExitHook} from 'exit-hook';
+
+const unsubscribe = asyncExitHook(async () => {
+	console.log('Exiting');
+}, {
+	minimumWait: 300
+});
+
+unsubscribe();
+```
+
+#### onExit
+
+Type: `Function` returns `Promise`
```

Comment:
Show the TS function signature instead.

Reply:
Also updated the original `exitHook` doc in readme to match formatting request